src/traces/parcoords/parcoords.js

```diff
@@ -557,7 +557,7 @@ module.exports = function(root, svg, styledData, layout, callbacks) {
                     .outerTickSize(2)
                     .ticks(wantedTickCount, d.tickFormat) // works for continuous scales only...
                     .tickValues(d.ordinal ? // and this works for ordinal scales
-                        sdom.map(function(d, i) {return texts && texts[i] || d;}) :
+                        sdom.map(function(d, i) {return texts && texts[i] !== undefined ? texts[i] : d;}) :
```

Comment:
With `parcoords` it's the situation you describe when the `tickvals` array is present and `ticktext` is missing: ticks will be placed possibly unevenly, while the ticks will be labeled with the numbers. The third axis is an example of that (0, 0.5, 1, 2, 3).

![image](https://user-images.githubusercontent.com/1548516/29222836-7d951128-7ec4-11e7-9232-b7c91b02122d.png)

Currently, the dimension in this case is considered categorical and `tickvals` do not get formatted. But as you say it'd probably better be, which is a one-liner change (+ test). The result looks like this, wondering if it should be included in this PR or a separate PR, or await further discussion.

![image](https://user-images.githubusercontent.com/1548516/29223239-27865cd6-7ec6-11e7-9d88-c6a1319eefa8.png)